### PR TITLE
Feature: Sponsor Offer Management

### PR DESCRIPTION
**Pull request by JustinCast**
# Sponsor offers management feature
- Includes the functionality to create﻿, update and delete
and offer.
- Lists all offers in a table and allows filtering

## How this should be tested?
- Clone this branch using github cli or download it in a zip
- Execute `make install` in the terminal in order to install dependencies
- Execute `make run` command in the terminal
- Go to `localhost:3000`

## Note
Codefactor throws some warning about camel case bugs. That couldn't be fixed. It's the way DB expect daba to be named.

*Please review the branch in your local environment and check if ev is ok*

### CODE REVIEW

File: hapi/src/api/user.api.js [hunk 1]
```diff
@@ -44,7 +44,6 @@ const SET_NAME = `
 
 const getOne = async (where = {}) => {
   const { user } = await hasuraUtils.request(GET_ONE, { where })
-
```
Comment (tetogomez) [inline hapi/src/api/user.api.js:47]: Why you delete that blank space?

Remember to add blank space after const or let declaration!

File: hapi/src/api/user.api.js [hunk 2]
```diff
@@ -44,7 +44,6 @@ const SET_NAME = `
...
   if (user && user.length > 0) {
```
Comment (tetogomez) [inline hapi/src/api/user.api.js:47]: this **if statement** could be a single line

`if (user && user.length > 0) return user[0]`

File: webapp/src/routes/OffersManagement/OffersManagement.js [hunk 3]
```diff
@@ -0,0 +1,234 @@
+import React, { useState, useEffect } from 'react'
+
```
Comment (tetogomez) [inline webapp/src/routes/OffersManagement/OffersManagement.js:2]: delete this blank space

File: webapp/src/routes/OffersManagement/OffersManagement.js [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+import MoreHorizIcon from '@material-ui/icons/MoreHoriz'
+import IconButton from '@material-ui/core/IconButton'
+import * as m from 'moment-timezone'
+import moment from 'moment'
+
+import {
+  GET_SPONSOR_OFFERS_QUERY,
+  UPDATE_OFFER_AVAILABILITY_MUTATION,
+  DELETE_OFFER_MUTATION
+} from '../../gql'
```
Comment (tetogomez) [inline webapp/src/routes/OffersManagement/OffersManagement.js:24]: add blank space

File: webapp/src/routes/OffersManagement/OffersManagement.js [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+  })
+
+  useEffect(() => {
+    const getOffers = async () => {
+      const { data } = await getSponsorOffers({
+        sponsor_id: 17
+      })
+
+      data && setOffers(data.offer)
+    }
```
Comment (tetogomez) [inline webapp/src/routes/OffersManagement/OffersManagement.js:84]: add blank space

File: webapp/src/routes/OffersManagement/OffersManagement.js [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+        break
+    }
+  }
+
+  useEffect(() => {
+    if (updateOfferResult) {
+      const newArr = [...offers]
+      const indexToUpdate = offers.findIndex(
+        (o) => o.id === updateOfferResult.returning[0].id
+      )
```
Comment (tetogomez) [inline webapp/src/routes/OffersManagement/OffersManagement.js:131]: same here

File: webapp/src/routes/OffersManagement/OffersManagement.js [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+    </FormControl>
+  )
+
+  const handleOpenClick = (offer) => {
+    setOpen(true)
+    setClickedOffer(offer)
+  }
+
+  const handleClose = (_event, reason) => {
+    if (reason === 'clickaway') {
```
Comment (tetogomez) [inline webapp/src/routes/OffersManagement/OffersManagement.js:166]: single line
`if (reason === 'clickaway') return`

File: webapp/src/routes/Profile/ProfilePageSponsor.js [hunk 8]
```diff
@@ -44,6 +44,8 @@ const useStyles = makeStyles((theme) => ({
 const ProfilePageSponsor = ({ profile }) => {
   const classes = useStyles()
 
+  console.log(profile)
```
Comment (tetogomez) [inline webapp/src/routes/Profile/ProfilePageSponsor.js]: unnecessary console log, please delete it!

**Review by tetogomez — COMMENTED** [review]
Check the comments

File: webapp/src/routes/OffersManagement/AddOffer.js [hunk 9]
```diff
@@ -0,0 +1,391 @@
...
+  ] = useLazyQuery(PROFILE_ID_QUERY, { fetchPolicy: 'network-only' })
+
+  useEffect(() => {
+    loadProfileID()
+  }, [loadProfileID])
+
+  const handleSubmit = () => {
+    const {
+      offer_type,
+      online_only,
```
Comment (AngeloCG97) [inline webapp/src/routes/OffersManagement/AddOffer.js:175]: hay algunas observaciones de codefactor.

**Comment by tetogomez** [comment]
also check the code factor comments.

**Comment by JustinCast** [comment]
> also check the code factor comments.

That couldn't be fixed. It's the way that DB expect data to be named

**Review by xavier506 — APPROVED** [review]
Nice Work!

**Review by JeanVegaD — APPROVED** [review]

**Comment by murillojorge** [comment]
@xavier506 should we go ahead and merge? or are those CodeFactor camelcase suggestions needed?

**Review by tetogomez — APPROVED** [review]

**Pull request merged**